### PR TITLE
infra: pgjdbc, migrate Maven -> Gradle

### PR DESCRIPTION
**Pull request by vlsi**
See https://github.com/pgjdbc/pgjdbc/pull/1726#issuecomment-594643435

when https://github.com/pgjdbc/pgjdbc/pull/1627 is merged this PR be ready for merge.

### CODE REVIEW

**Comment by rnveach** [comment]
````
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':postgresql:checkstyleMain'.
> Could not resolve all files for configuration ':postgresql:checkstyle'.
   > Could not find com.puppycrawl.tools:checkstyle:8.31-SNAPSHOT.
     Searched in the following locations:
       - file:/root/.m2/repository/com/puppycrawl/tools/checkstyle/8.31-SNAPSHOT/maven-metadata.xml
       - file:/root/.m2/repository/com/puppycrawl/tools/checkstyle/8.31-SNAPSHOT/checkstyle-8.31-SNAPSHOT.pom
       - https://repo.maven.apache.org/maven2/com/puppycrawl/tools/checkstyle/8.31-SNAPSHOT/maven-metadata.xml
       - https://repo.maven.apache.org/maven2/com/puppycrawl/tools/checkstyle/8.31-SNAPSHOT/checkstyle-8.31-SNAPSHOT.pom
     Required by:
         project :postgresql

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
````

**Comment by vlsi** [comment]
@rnveach , thanks.
Do you have access to the filesystem?

The build log says it attempted to resolve Checkstyle snapshot from Maven local repository, however, it looks like the snapshot is not there:

>     Searched in the following locations:
>       - file:/root/.m2/repository/com/puppycrawl/tools/checkstyle/8.31-SNAPSHOT/maven-metadata.xml
>       - file:/root/.m2/repository/com/puppycrawl/tools/checkstyle/8.31-SNAPSHOT/checkstyle-8.31-SNAPSHOT.pom

Can you please check if Checkstyle is installed properly?

**Comment by rnveach** [comment]
https://github.com/checkstyle/checkstyle/blob/master/wercker.yml#L58
the first thing wercker does is install the snapshot version for use with every run.

> [INFO] --- maven-install-plugin:2.5.2:install (default-install) @ checkstyle ---
[INFO] Installing /pipeline/source/target/checkstyle-8.31-SNAPSHOT.jar to /pipeline/cache/com/puppycrawl/tools/checkstyle/8.31-SNAPSHOT/checkstyle-8.31-SNAPSHOT.jar
[INFO] Installing /pipeline/source/pom.xml to /pipeline/cache/com/puppycrawl/tools/checkstyle/8.31-SNAPSHOT/checkstyle-8.31-SNAPSHOT.pom
[INFO] Installing /pipeline/source/target/checkstyle-8.31-SNAPSHOT-tests.jar to /pipeline/cache/com/puppycrawl/tools/checkstyle/8.31-SNAPSHOT/checkstyle-8.31-SNAPSHOT-tests.jar

**Comment by pzygielo** [comment]
Should this
https://github.com/checkstyle/checkstyle/blob/a8297ea36fbef327817d7604eef245d0150ca971/wercker.yml#L8 be considered/set for new gradle call as well?

It seems maven installs to `/pipeline/cache` but gradle is looking in `/root/.m2/repository`.

**Comment by vlsi** [comment]
I see there's https://github.com/checkstyle/checkstyle/blob/a8297ea36fbef327817d7604eef245d0150ca971/wercker.yml#L18

Is that command executed?

Frankly speaking, the command looks suspicious to me. I guess it should be `ln -s ${WERCKER_CACHE_DIR} /root/.m2`

I've pushed a commit that would list files in `/root/.m2`.
Can you please show the output from the latest wercker build?

**Comment by pzygielo** [comment]
My fault,
> I see there's
> 
> https://github.com/checkstyle/checkstyle/blob/a8297ea36fbef327817d7604eef245d0150ca971/wercker.yml#L18

I didn't :frowning_face:

**Comment by vlsi** [comment]
wercker shows `401` for me.

Is there a way I can get permissions to watch CI logs?

It is really painful to debug CI jobs which show 401  :-/

/cc @rnveach , @romani  ?

File: .ci/wercker.sh [hunk 1]
```diff
@@ -52,10 +52,16 @@ no-error-pgjdbc)
   CS_POM_VERSION=$(mvn -e -q -Dexec.executable='echo' -Dexec.args='${project.version}' \
                      --non-recursive org.codehaus.mojo:exec-maven-plugin:1.3.1:exec)
   echo CS_version: ${CS_POM_VERSION}
+  ls -l /root/.m2
+  ls -l /root/.m2/repository
+  ls -l /root/.m2/repository/com/puppycrawl/tools/checkstyle
+  ls -l /root/.m2/repository/com/puppycrawl/tools/checkstyle/8.31-SNAPSHOT
```
Comment (rnveach) [inline .ci/wercker.sh]: > CS_version: 8.31-SNAPSHOT
total 12
-rw-r--r--. 1 root root  129 Mar  9 15:24 copy_reference_file.log
drwxr-xr-x. 2 root root 4096 Mar  9 15:24 repository
-rw-r--r--. 1 root root  327 Mar  9 15:24 settings-docker.xml
total 0
lrwxrwxrwx. 1 root root 15 Mar  9 15:24 cache -> /pipeline/cache
ls: cannot access '/root/.m2/repository/com/puppycrawl/tools/checkstyle': No such file or directory

As we discussed earlier, checkstyle isn't stored into the `.m2` folder.

**Comment by vlsi** [comment]
@rnveach, can you please help me with the new logs? (a04eb6a and 408ffa7)
That is the output of `ls ..` commands and the output of `./gradlew`. I think it should pick `GRADLE_OPTS`, and it should pick `-Dmaven.repo.local`

**Comment by rnveach** [comment]
@vlsi Wercker passed.

**Comment by romani** [comment]
looks good, thanks a lot for update.

**Comment by rnveach** [comment]
@vlsi PR is still listed as draft. Will you be finalizing this at some time? If so, please rebase this on the latest master to ensure it will not conflict with other changes being made.

**Comment by vlsi** [comment]
The PR is still draft as the upstream pgjdbc is not migrated to Gradle yet.
My estimation is the switch will happen on the next week or so.

**Comment by vlsi** [comment]
@romani , @rnveach , pgjdbc has switched from Maven to Gradle, so please update wercker configuration.

**Comment by romani** [comment]
this is become an issue for master build of wercker - https://app.wercker.com/checkstyle/checkstyle/runs/build/5e81e9a8765d2a000763f7ce?step=5e81ea05ee64ed00087ab0fc

This PR should be expedited.

**Comment by romani** [comment]
wercker is passed, all other major validation are passed, I merged without waiting for all pass state.

**Comment by romani** [comment]
@vlsi , thanks a lot for update!

**Comment by romani** [comment]
@vlsi , I am not sure what happened but pgjdbc is still on maven in master branch.
I did revert of your commit - https://github.com/checkstyle/checkstyle/commit/522795be5e85e6b5ac87ac1c12c55711950ce6f5, to unblock all our PRs.

Please explain what happened.

**Comment by vlsi** [comment]
@romani , there was a regression in the previous release, so we did a rollback and re-released pgjdbc. Sorry for the inconvenience.

Now maven->gradle is re-applied to pgjdbc.

PS. We don't expect to rollback once again, however, it would probably make sense to use shell-script-like integration to avoid disruptions when the command line to run checkstyle changes.

**Comment by rnveach** [comment]
@vlsi Is there a central issue to track conversion to gradle so we can watch and know if something goes wrong again?

**Comment by vlsi** [comment]
It is not related to Gradle.

The issue was with a couple of fixes that were released, however, one of them introduced regressions, and another was too big to include into a minor release.

So we tried to play safe and reverted the whole thing.
Apparently, we did not expect it would ripple in the failures elsewhere.

**Comment by romani** [comment]
we switched back to gradle , wercker is passing - https://app.wercker.com/checkstyle/checkstyle/runs/build/5e849f9c7b63df001af7d8fb?step=5e849fdc1ccc9800081d78ce

@vlsi , thanks a lot for help.

**Pull request merged**